File: hzy-admin-client/src/service/development_tool/monitor_ef_core_service.js
import { post } from '@/scripts/request';

const controllerName = "admin/MonitorEFCore";

export default {
    /**
     * 获取 EFCore 监控上下文
     *
     */
    getEFCoreMonitorContext() {
        return post(`${controllerName}/GetEFCoreMonitorContext/`, null, false);
    },
    /**
     * 获取 EFCore Sql 监控上下文 耗时排行榜
     *
     */
    getTimeConsumingRanking() {
        return post(`${controllerName}/GetTimeConsumingRanking/`, null, false);
    },
    /**
    * 获取 EFCore Sql 监控上下文 最新数据
    *
    */
    getNewest() {
        return post(`${controllerName}/GetNewest/`, null, false);
    },

};

File: hzy-admin-server/HZY.Services.Admin/DevelopmentTool/MonitorEFCore/Impl/MonitorEFCoreService.cs
using HZY.Framework.EntityFrameworkRepositories.Monitor;
using HZY.Framework.EntityFrameworkRepositories.Monitor.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HZY.Services.Admin.DevelopmentTool.MonitorEFCore.Impl
{
    /// <summary>
    /// 监控 EFCore 服务
    /// </summary>
    public class MonitorEFCoreService : IMonitorEFCoreService
    {
        public EntityFrameworkRepositoriesMonitorContext GetEFCoreMonitorContext()
        {
            return EntityFrameworkRepositoriesMonitorCache.Context;
        }

        public List<EntityFrameworkRepositoriesMonitorSqlContext> GetNewest()
        {
            return EntityFrameworkRepositoriesMonitorCache.SqlContext.OrderByDescending(w => w.Time).Take(20).ToList();
        }

        public List<EntityFrameworkRepositoriesMonitorSqlContext> GetTimeConsumingRanking()
        {
            return EntityFrameworkRepositoriesMonitorCache.SqlContext.OrderByDescending(w => w.ElapsedMilliseconds).Take(20).ToList();
        }



    }
}
